src/ring/middleware/oauth2.clj

```diff
@@ -19,6 +19,7 @@
 (defn- authorize-uri [profile request state]
   (str (:authorize-uri profile)
        (if (.contains ^String (:authorize-uri profile) "?") "&" "?")
+       (request :query-string)
```

Comment:
a little unsure how to handle cases where there is no query string, though the tests are green so perhaps this is enough?

Reply:
I'd suggest changing this to: `(:query-string request)`. In general that's more robust.